src/helpers.ts

```diff
@@ -139,6 +144,44 @@ export function getCourseInfo(SOCObject: WebsocResponse) {
     return courseInfo;
 }
 
+export interface ZotCourseResponse {
+    codes: string[];
+    customEvents: RepeatingCustomEvent[];
+}
+export async function queryZotCourse(schedule_name: string) {
+    const url = new URL(ZOTCOURSE_ENDPOINT);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

Comment:
Can you explain why we had to disable a linter rule here?

Reply:
tbh I need help with this and the following. TS raises errors since the response value has no type. 
I couldn't come up with a solution other than writing brand types that are only for these temporary variables.
What can be a good practice to solve this?

Reply:
The types should work themselves out if you define an interface for the entire `fetch` result. Looks like you already defined a interface called `ZotCourseResponse` for the return type of `queryZotcourse`, but I'd recommend using it to typecast your fetch request instead and let typescript infer the return type of your function. See this example from `News.tsx`: 
![image](https://user-images.githubusercontent.com/48658337/232926973-368da4cd-77e3-4d21-aff3-0ce846f9a60e.png)

Reply:
IDK why Eslint stops complaining about them when I remove those disabling lines now.